### ISSUE TITLE
Error spawning pyls unless I am root

### ISSUE THREAD

**Issue by eyalk5** [onivim/oni#2726]
**Oni Version:**
0.3.4
**Neovim Version (Linux only):**
**Operating System:**
Mac OS  10.13.2
**Issue:**
When I open a python file, language server is not avaliable for python.
That is unless I am root. I get the error : spawn pyls ENONET.
When I am root, it works fine.
I have installed pyls regularly and can run it without root from cmd line.

**Comment by eyalk5** [comment]
![image](https://user-images.githubusercontent.com/23480511/53092521-a6c89e00-351d-11e9-82da-2ebd609882f7.png)

**Comment by TalAmuyal** [comment]
Can you run pyls from the command-line w/o root and with no configs that
will be missing for oni?

**Comment by eyalk5** [comment]
Indeed. When I simply run pyls from command line, it seems to be working.
eyalkarni@xxxx:~$ pyls -v
2019-02-25 01:32:05,154 UTC - INFO - pyls.python_ls - Starting PythonLanguageServer IO language server

**Comment by TalAmuyal** [comment]
@eyalk5 I'm not completely sure about it, but I do think it's pathing/env-var issue.
How about the following in your configs?
```json
  "language.python.languageServer": {"command": "python", "arguments": ["-m", "pyls"]}
```

For me it works while explicitly stating `python3`: `"language.python.languageServer": {"command": "python3", "arguments": ["-m", "pyls"]}`

**Comment by eyalk5** [comment]
Thank you. It seems to be working.

**Comment by eyalk5** [comment]
OK, it doesn't set the change. When I tried to set (in activate):
```
 oni.language.python.languageServer.command="python"
  oni.language.python.languageServer.arguments= ["-m", "pyls"]

```
then it raises an error when trying to upgrade config.js. And keeps using pyls.

When I try to edit config.js :
```

exports.configuration = {
  "language.python.languageServer": {"command": "python", "arguments": ["-m", "pyls"]}
    //add custom config here, such as
    "ui.colorscheme": "nord",
    //"oni.useDefaultConfig": true,
    //"oni.bookmarks": ["~/Documents"],
    //"oni.loadInitVim": false,
    //"editor.fontSize": "12px",
    //"editor.fontFamily": "Monaco",
    // UI customizations
    "ui.animations.enabled": true,
    "ui.fontSmoothing": "auto"
};
```
It simply ignores it. 

I don't understand. 
Thanks for the help so far!

**Comment by eyalk5** [comment]
![image](https://user-images.githubusercontent.com/23480511/54074406-be808000-429a-11e9-8209-5b3c488215a9.png)

(The error I get)

**Comment by TalAmuyal** [comment]
Not sure what's the source of the issue, could you try [this](https://gist.github.com/TalAmuyal/87621859a96dc07e680a89d5f48763b8) config?
